src/backend/distributed/commands/table.c

```diff
@@ -476,6 +501,41 @@ PreprocessAlterTableMoveAllStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PreprocessAlterTableSchemaStmt is executed before the statement is applied to the local
+ * postgres instance.
+ *
+ * In this stage we can prepare the commands that will alter the schemas of the shards.
+ */
+List *
+PreprocessAlterTableSchemaStmt(Node *node, const char *queryString)
+{
+	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
+	Assert(stmt->objectType == OBJECT_TABLE);
+
+	if (stmt->relation == NULL)
+	{
+		return NIL;
+	}
+	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt,
+														  stmt->missing_ok);
+	Oid relationId = address.objectId;
+
+	/* first check whether a distributed relation is affected */
+	if (!OidIsValid(relationId) || !IsDistributedTable(relationId))
+	{
+		return NIL;
+	}
+	DDLJob *ddlJob = palloc0(sizeof(DDLJob));
+	QualifyTreeNode((Node *) stmt);
+	ddlJob->targetRelationId = relationId;
+	ddlJob->concurrentIndexCmd = false;
+	ddlJob->commandString = DeparseTreeNode((Node *) stmt);
+	ddlJob->taskList = DDLTaskList(relationId, DeparseTreeNode((Node *) stmt));
```

Comment:
Let's reuse the result of the first deparse.
Deparsing is quite resource intensive and pressing on memory, where possible we should reuse the same output.